torch/csrc/jit/passes/quantization.cpp

```diff
@@ -130,47 +130,67 @@ class InsertObserversHelper {
       const script::Module& module,
       const std::string& method_name);
 
-  // Values that are the output of GetAttr[name="weight"] and
-  // GetAttr[name="bias"] will be propagated from parent method call to the
-  // child graph
-  void propagateValues(Node* n, std::shared_ptr<Graph>& graph);
-
   const ModuleQConfigMap& module_qconfig_map_;
   // Values we want to skip observing, used to skip values in
   // the middle of the ops that are supposed to be fused, e.g.
   // the output value of conv in the conv - relu pattern
   std::unordered_set<Value*> values_to_skip_;
-  // Values that are the output of GetAttr[name="weight"] and they
-  // will be propagated through the function call hierarchy
-  std::unordered_set<Value*> weight_values_;
-  // Values that are the output of GetAttr[name="bias"] and they
-  // will be propagated through the function call hierarchy
-  std::unordered_set<Value*> bias_values_;
   // Unique id generator for observer module, used for generating
   // unique observer names when we insert observer module, we
   // record the current unique id used to avoid incrementing from 0
   // every time to find a unique id.
   int uid_ = 0;
 };
 
+bool isBiasOfConvOrLinear(Value* v) {
+  for (const Use& u : v->uses()) {
+    if (u.user->kind() == Symbol::aten("conv2d")) {
+      if (v == u.user->inputs().at(2)) {
+        return true;
+      }
+    } else if (u.user->kind() == prim::CallFunction) {
+      auto func_name = getFuncName(u.user->inputs()[0]);
+      if (func_name == "linear" && v == u.user->inputs().at(3)) {
+        return true;
+      }
+    }
+  }
+  return false;
+}
+
+bool isWeightOfConvOrLinear(Value* v) {
+  for (const Use& u : v->uses()) {
```

Comment:
Do we actually allow multiple uses? How would we quantize it if it's used in several places? If we don't support this case, we should assert that there is only one use.

Reply:
we don't support multiple uses, and for weight we probably don't need to. I'll add an assert.